lib/block.js

```diff
@@ -34,35 +33,14 @@ export default class Block {
     return this.getMerkleTree().getHexRoot();
   }
 
-  unsignedHeader(payload) {
-    payload.write(this.parent.replace('0x', ''), 0, 'hex');
-    writeUint64(payload, this.height, 32);
-    payload.write(this.merkleRoot().replace('0x', ''), 40, 'hex');
+  header(payload) {
+    writeUint64(payload, this.height, 0);
+    payload.write(this.merkleRoot().replace('0x', ''), 8, 'hex');
     return payload;
   }
 
-  sigHash() {
-    return ethUtil.sha3(this.unsignedHeader(Buffer.alloc(72, 0)));
-  }
-
-  sign(privKey) {
-    const privBuf = toBuffer(privKey);
-    const sig = ethUtil.ecsign(this.sigHash(), privBuf);
-    this.v = sig.v;
-    this.r = sig.r;
-    this.s = sig.s;
-    return [sig.v, bufferToHex(sig.r), bufferToHex(sig.s)];
-  }
-
   hash() {
-    if (!this.v || !this.r || !this.s) {
-      throw new Error('not signed yet');
-    }
-    const payload = this.unsignedHeader(Buffer.alloc(137, 0));
-
-    payload.writeUInt8(this.v, 72);
-    this.r.copy(payload, 73);
-    this.s.copy(payload, 105);
+    const payload = this.header(Buffer.alloc(48, 0));
```

Comment:
why 48 bytes? i see 32 for merkle root, and 8 for height => 40

Reply:
Because I'm a god of arithmetic :-)